mlflow/server/handlers.py

```diff
@@ -16,29 +17,42 @@
     UpdateRun, LogMetric, LogParam, SetTag, ListExperiments, \
     DeleteExperiment, RestoreExperiment, RestoreRun, DeleteRun, UpdateExperiment, LogBatch
 from mlflow.store.artifact_repository_registry import get_artifact_repository
-from mlflow.tracking.utils import _is_database_uri, _is_local_uri
+from mlflow.tracking.utils import TrackingStoreRegistry
 from mlflow.utils.proto_json_utils import message_to_json, parse_dict
 from mlflow.utils.search_utils import SearchFilter
 from mlflow.utils.validation import _validate_batch_log_api_req
 
 _store = None
 
 
+def _get_file_store(store_uri, artifact_uri):
+    from mlflow.store.file_store import FileStore
+    return FileStore(store_uri, artifact_uri)
+
+
+def _get_sqlalchemy_store(store_uri, artifact_uri):
+    from mlflow.store.sqlalchemy_store import SqlAlchemyStore
+    return SqlAlchemyStore(store_uri, artifact_uri)
+
+
+_tracking_store_registry = TrackingStoreRegistry()
+_tracking_store_registry.register('', _get_file_store)
+_tracking_store_registry.register('file', _get_file_store)
+for scheme in DATABASE_ENGINES:
+    _tracking_store_registry.register(scheme, _get_sqlalchemy_store)
+
+
 def _get_store(backend_store_uri=None, default_artifact_root=None):
     from mlflow.server import BACKEND_STORE_URI_ENV_VAR, ARTIFACT_ROOT_ENV_VAR
     global _store
     if _store is None:
-        store_dir = backend_store_uri or os.environ.get(BACKEND_STORE_URI_ENV_VAR, None)
+        store_uri = backend_store_uri or os.environ.get(BACKEND_STORE_URI_ENV_VAR, None)
         artifact_root = default_artifact_root or os.environ.get(ARTIFACT_ROOT_ENV_VAR, None)
-        if _is_database_uri(store_dir):
-            from mlflow.store.sqlalchemy_store import SqlAlchemyStore
-            _store = SqlAlchemyStore(store_dir, artifact_root)
-        elif _is_local_uri(store_dir):
-            from mlflow.store.file_store import FileStore
-            _store = FileStore(store_dir, artifact_root)
-        else:
+        try:
+            _store = _tracking_store_registry.get_store(store_uri, artifact_root)
+        except MlflowException:
```

Comment:
This is a bit dodgy since it can mask other failures. But the `TrackingStoreRegistry` error is not user-friendly here. Would it make sense to change the error on the  `TrackingStoreRegistry` to something like "Unexpected backend store URI scheme: {}" as a compromise?

Reply:
+1 this suggestion.
This registry is a developer concept and unrelated to what the user is asking.